src/setup/node.rs

```diff
@@ -67,21 +72,22 @@ impl NodeBuilder {
 
         let conf = NodeConfig::default();
         let meta = NodeMetaData::new(setup_path)?;
-        Ok(Self { conf, meta })
+
+        // We always expect to get indices in increasing order when popping the vector.
+        let stateful_nodes_pool = (0..STATEFUL_NODES_COUNT).rev().collect::<Vec<usize>>();
+
+        Ok(Self {
+            conf,
+            meta,
+            stateful_nodes_pool,
+        })
     }
 
     /// Creates new [NodeBuilder] which can handle stateful nodes.
     pub fn stateful() -> anyhow::Result<Self> {
         Ok(Self::stateless()
             .expect("Failed to create a node builder")
-            .network_id(TESTNET_NETWORK_ID)
-            .validator_token(get_validator_token(0))
-            .add_args(vec![
-                "--valid".into(),
-                "--quorum".into(),
-                "1".into(),
-                "--load".into(),
-            ]))
```

Comment:
Unfortunately, I had to remove most of these from here.
And then `add_args` became useless.